roofit/roofitcore/src/FitHelpers.cxx

```diff
@@ -66,7 +67,19 @@ constexpr int extendedFitDefault = 2;
 /// \param[in] data The dataset that was used for the fit.
 int calcAsymptoticCorrectedCovariance(RooAbsReal &pdf, RooMinimizer &minimizer, RooAbsData const &data)
 {
-   // Calculated corrected errors for weighted likelihood fits
+  RooFormulaVar logpdf("logpdf", "-log(pdf)", "-log(@0)", pdf);
```

Comment:
Can't we just use [RooAbsPdf::getLogVal()](https://root.cern.ch/doc/master/classRooAbsPdf.html#af9408e8c291b61fb781f4ee5b923021c)?

Reply:
I believe RooDerivative would not take that as an argument, so I do not think this is possible directly for my use case. I am happy if someone with more internal knowledge on roofit has a concrete suggestion how to get a RooAbsPdf which is the log of an existing pdf.

Reply:
Note that I have changed the sign here -log() -> +log(). While the overall sign does not matter, the relative sign to the extended term is important. Since I derive +log(Nexpected), the sign here needs to be positive as well.